packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx

```diff
@@ -222,8 +222,13 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
 
   @autobind
   private _onGotoToday() {
-    this._navigateDay(this.props.today!);
+    let { onSelectDate, today } = this.props;
+    this._navigateDay(today!);
     this._focusOnUpdate = true;
+
+    if (onSelectDate) {
```

Comment:
Instead of this, should we just always call this._onSelectDate and have the behavior be to always set selectedDate regardless whether there is a callback or not? BTW, there are two open issues on github related to this:
- https://github.com/OfficeDev/office-ui-fabric-react/issues/2516
- https://github.com/OfficeDev/office-ui-fabric-react/issues/1850

Reply:
Per offline chat: a better approach would be to call onSelectItem:

  private _onGotoToday() {
    let { onSelectDate, dateRangeType, firstDayOfWeek, today } = this.props;
    let dates = getDateRangeArray(today!, dateRangeType!, firstDayOfWeek!);
    this._onSelectDate(today!, dates);
  } 

This solves a couple problems:
- Does not rely on side effects, i.e., with the original solution, go to today doesnt work if there consumer doesn't give you an onSelectDate callback
- Invokes the callbacks with the correct dates set for the current date range